### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by viroot27**
**Summary**
Writing new documentation for all the modules in the `x/` folder. Here's the Readme for `x/epochs/` primarily taken from some of Aidan's earlier work.

I'm getting rid of the `x/epochs/spec/` folder, but I started the README off by copy pasting the content from all the files in that folder to the new README and editing from there

### CODE REVIEW

File: x/epochs/README.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+```protobuf
+service Query {
+  // EpochInfos provide running epochInfos
+  rpc EpochInfos(QueryEpochsInfoRequest) returns (QueryEpochsInfoResponse) {}
+  // CurrentEpoch provide current epoch of specified identifier
+  rpc CurrentEpoch(QueryCurrentEpochRequest) returns (QueryCurrentEpochResponse) {}
+}
+```
+---
+## Future Improvements
```
Comment (viroot27) [inline x/epochs/README.md:149]: Do we want to keep this? Is this still relevant?

**Comment by riley-stride** [comment]
Good stuff! Helpful to have this all documented. A good next step would be to add an explanation to this README of how our stride specific epochly logic works. Can be a separate PR!

**Review by riley-stride — APPROVED** [review]
Looks good to merge! Good job on the description, it made this clear/easy to review.

**Review by asalzmann — APPROVED** [review]
:LGTM: ! (looks good to me)

**Pull request merged**